src/event.rs

```diff
@@ -253,6 +253,25 @@ pub enum Event<T: 'static> {
     ///
     /// - **macOS / Wayland / Windows / Orbital:** Unsupported.
     MemoryWarning,
+
+    /// Emitted when the application has received a reopen request from OS.
+    ///
+    /// ## Platform-specific
+    ///
+    /// ### macOS
+    ///
+    /// On macOS, the `Reopen` event is emitted in response to an [`applicationShouldHandleReopen`]
+    /// callback, which is typically called when the user clicks on the application's icon in the Dock.
+    /// The application should create a new window if it doesn't have any, or bring the
+    /// existing windows to the front. `has_visible_windows` is `true` if the application has
+    /// at least one visible window.
```

Comment:
Maybe document it more in Winit terms?

```suggestion
    /// Usually, the user would expect you to create a new window if there isn't any.
```